x-pack/plugins/lens/public/indexpattern_datasource/indexpattern_suggestions.ts

```diff
@@ -117,6 +117,23 @@ export function getDatasourceSuggestionsForField(
   }
 }
 
+// Called when the user navigates from Discover to Lens (Visualize button)
+export function getDatasourceSuggestionsForVisualizeField(
+  state: IndexPatternPrivateState,
+  indexPatternId: string,
+  fieldName: string
+): IndexPatternSugestion[] {
+  const layers = Object.keys(state.layers);
+  const layerIds = layers.filter((id) => state.layers[id].indexPatternId === indexPatternId);
+  // Identify the field by the indexPatternId and the fieldName
+  const indexPattern = state.indexPatterns[indexPatternId];
+  const field = indexPattern.fields.find((fld) => fld.name === fieldName);
+
+  if (layerIds.length === 0 || !field) return [];
+  const layerId = layerIds[0];
+  return getEmptyLayerSuggestionsForField(state, layerId, indexPatternId, field);
+}
```

Comment:
This is an improvement that you could consider, but I don't think it's required: Other suggestions are looking for the "emptiest" layer for a specific index pattern, not just the first layer.